tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,696 @@
+    pub pii_values: Vec<String>,
+}
+
+impl Cli {
+    fn write_pii<W: Write>(&self, writer: W, metrics: &[Pii]) -> anyhow::Result<()> {
+        match self.format {
+            SchemaFormat::Json => serde_json::to_writer_pretty(writer, metrics)?,
+            SchemaFormat::Yaml => serde_yaml::to_writer(writer, metrics)?,
+        };
+
+        Ok(())
+    }
+
+    pub fn run(self) -> anyhow::Result<()> {
+        let rust_file_paths = {
+            let path = self
+                .path
+                .clone()
+                .unwrap_or_else(|| std::env::current_dir().unwrap());
+
+            find_rs_files(&path)
+        };
+
+        // Before we can iterate over the pii fields properly, we make a mapping between all
+        // paths to types and their AST node.
+        FileSyntaxVisitor::fill_pii_hashmaps(&rust_file_paths);
+
+        {
+            // Sets the pii_values from Cli to a global variable so it can be accessed in the
+            // 'Visit' trait.
+            let mut guard = PII_VALUES.lock().unwrap();
+            guard.clear();
+            guard.extend(self.pii_values.clone());
+        }
+
+        find_pii_fields(self.item.as_deref());
+
+        let output_vec = get_pii_fields_output("".to_string());
+
+        match self.output {
+            Some(ref path) => self.write_pii(File::create(path)?, &output_vec)?,
+            None => self.write_pii(std::io::stdout(), &output_vec)?,
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Copy, Debug, ValueEnum, Default)]
+pub enum SchemaFormat {
+    #[default]
+    Json,
+    Yaml,
+}
+
+/// Takes in the path to a rust file and gives back the way you'd refer to it in a use-statement.
+/// e.g. "/Users/tor/prog/rust/relay/relay-general/src/protocol/types.rs -> relay_general::protocol"
+fn rust_file_to_use_path(file_path: &Path) -> String {
+    let crate_name = {
+        let file_str = file_path.as_os_str().to_str().unwrap();
+        let src_index = file_str.find("/src/").unwrap();
+        let back_index = file_str[..src_index].rfind('/').unwrap() + 1;
+        file_str.split_at(src_index).0.split_at(back_index).1
+    };
+    let parent_dir = file_path.parent().unwrap();
+    let file_stem = file_path.file_stem().unwrap().to_string_lossy().to_string();
+
+    let is_module = is_file_module(file_path);
+
+    let mut module_path = parent_dir
+        .components()
+        .map(|part| part.as_os_str().to_string_lossy().to_string())
+        .filter(|part| part != "src")
+        .collect::<Vec<_>>();
+
+    if is_module {
+        module_path.push(file_stem);
+    }
+
+    let mut use_path = module_path.join("::");
+    use_path = use_path
+        .split_at(use_path.find(crate_name).unwrap())
+        .1
+        .replace('-', "_");
+
+    use_path
+}
+
+/// Checks if a file is a module, which is needed to convert a file path to a module path
+fn is_file_module(file_path: &Path) -> bool {
+    let parent_dir = file_path.parent().unwrap();
```

Comment:
Instead of using `unwrap`s , which does not really matter for tooling , but we still wanna enforce best practices, return the `Result` and propagate the error to the `main` (maybe), where the error could be handled and properly printed / showed to the caller, with proper error message. 
You could also use `anyhow` crate for this and add the `context` to each error to provide even more information on why the error happens. 


Same goes about all the `unwrap`s in this and other functions.

Reply:
@TBS1996  why did you resole this without addressing the comment? 
Can we, please, remove unwraps and return `Result` here?

Reply:
my bad, I did it on a different function and didn't see that this one also would require a change